### PR TITLE
sensors : fix bad pin value when disabling interruption

### PR DESCRIPTION
**Pull request by lpasteau**
In gpio callback function of some sensor drivers, the device structure in parameter is about the gpio and not the sensor.

Signed-off-by: laurence pasteau <laurence.pasteau@stimio.fr>

### CODE REVIEW

File: drivers/sensor/adxl372/adxl372_trigger.c [hunk 1]
```diff
@@ -51,14 +51,14 @@ static void adxl372_thread_cb(void *arg)
 				     GPIO_INT_EDGE_TO_ACTIVE);
 }
 
-static void adxl372_gpio_callback(struct device *dev,
+static void adxl372_gpio_callback(struct device *dev_port,
```
Comment (jfischer-phytec-iot) [inline drivers/sensor/adxl372/adxl372_trigger.c:54]: This change is not necessary, it just makes hard to read your patch. Please remove this change everywhere.

File: drivers/sensor/adxl372/adxl372_trigger.c [hunk 2]
```diff
@@ -51,14 +51,14 @@ static void adxl372_thread_cb(void *arg)
...
 				  struct gpio_callback *cb, uint32_t pins)
 {
 	struct adxl372_data *drv_data =
 		CONTAINER_OF(cb, struct adxl372_data, gpio_cb);
-	const struct adxl372_dev_config *cfg = dev->config_info;
+	const struct adxl372_dev_config *cfg = drv_data->dev->config_info;
 
-	gpio_pin_interrupt_configure(drv_data->gpio, cfg->int_gpio,
+	gpio_pin_interrupt_configure(dev_port, cfg->int_gpio,
```
Comment (jfischer-phytec-iot) [inline drivers/sensor/adxl372/adxl372_trigger.c:61]: It seems do not change or fix any behavior? Please remove this change.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

**Comment by lpasteau** [comment]
@jfischer-phytec-iot the correction is done !

File: drivers/sensor/lis2dw12/lis2dw12_trigger.c [hunk 3]
```diff
@@ -128,7 +128,7 @@ static int lis2dw12_handle_drdy_int(struct device *dev)
 static int lis2dw12_handle_single_tap_int(struct device *dev)
 {
 	struct lis2dw12_data *data = dev->driver_data;
-	sensor_trigger_handler_t handler = data->tap_handler;;
+	sensor_trigger_handler_t handler = data->tap_handler;
```
Comment (jfischer-phytec-iot) [inline drivers/sensor/lis2dw12/lis2dw12_trigger.c:131]: This changes are out of place here, please move it to a separate commit. Thanks.

**Review by jfischer-no — COMMENTED** [review]
Thanks, just one minor comment. I can see there are more drivers affected, would you like to fix them too?

File: drivers/sensor/lis2dw12/lis2dw12_trigger.c (hunk 3 above)
Comment (lpasteau) [reply drivers/sensor/lis2dw12/lis2dw12_trigger.c:131]: In a separate commit of the same pull request ?

**Comment by lpasteau** [comment]
> 
> 
> Thanks, just one minor comment. I can see there are more drivers affected, would you like to fix them too?

Which drivers ? I thought I made the correction for all affected sensors.

File: drivers/sensor/lis2dw12/lis2dw12_trigger.c (hunk 3 above)
Comment (MaureenHelm) [reply drivers/sensor/lis2dw12/lis2dw12_trigger.c:131]: Yes

**Review by MaureenHelm — COMMENTED** [review]
@lpasteau thanks for the fixes!

@avisconti please review

**Comment by jfischer-phytec-iot** [comment]
> > Thanks, just one minor comment. I can see there are more drivers affected, would you like to fix them too?
> 
> Which drivers ? I thought I made the correction for all affected sensors.

drivers/sensor/amg88xx
drivers/sensor/adxl362
drivers/sensor/adxl372 :white_check_mark: 
drivers/sensor/bmg160
drivers/sensor/bmi160
drivers/sensor/fxas21002
drivers/sensor/fxos8700
drivers/sensor/hmc5883l
drivers/sensor/iis2dlpc
drivers/sensor/iis2mdc :white_check_mark: 
drivers/sensor/iis3dhhc :white_check_mark: 
drivers/sensor/isl29035
drivers/sensor/ism330dhcx :white_check_mark: 
drivers/sensor/lis2dh
drivers/sensor/lis2ds12 :white_check_mark: 
drivers/sensor/lis2dw12 :white_check_mark: 
drivers/sensor/lis2mdl :white_check_mark: 
drivers/sensor/lis3mdl
drivers/sensor/lps22hh :white_check_mark: 
drivers/sensor/lsm6dsl
drivers/sensor/lsm6dso :white_check_mark: 
drivers/sensor/stts751 :white_check_mark: 
drivers/sensor/sx9500
:smile: 

@MaureenHelm Maybe we should remove all the CONFIG_*_TRIGGER_OWN_THREAD?

**Comment by lpasteau** [comment]
> 
> 
> > > Thanks, just one minor comment. I can see there are more drivers affected, would you like to fix them too?
> > 
> > 
> > Which drivers ? I thought I made the correction for all affected sensors.
> 
> 🤦‍♂️ was my mistake
> 
> @MaureenHelm Maybe we should remove all the CONFIG_*_TRIGGER_OWN_THREAD?

This config is very useful sometimes in order to separate correctly one sensor.

File: drivers/sensor/lis2dw12/lis2dw12_trigger.c (hunk 3 above)
Comment (lpasteau) [reply drivers/sensor/lis2dw12/lis2dw12_trigger.c:131]: Done !

**Review by jfischer-no — APPROVED** [review]

**Comment by jfischer-phytec-iot** [comment]
> > @MaureenHelm Maybe we should remove all the CONFIG_*_TRIGGER_OWN_THREAD?
> 
> This config is very useful sometimes in order to separate correctly one sensor.

That probably could be done differently. IMO a global option to select either common sensor workqueue or system workqueue would be better solution.

**Review by MaureenHelm — APPROVED** [review]
